src/query/execute_statement.c

```diff
@@ -14809,12 +14809,67 @@ do_replicate_statement (PARSER_CONTEXT * parser, PT_NODE * statement)
     {
       PT_PRINT_VALUE_FUNC saved_func = parser->print_db_value;
       int saved_custom_print = parser->custom_print;
+      PARSER_VARCHAR **host_val;
+      char *sbr_text, *sql_text;
+      int i, n, nth;
+      int sql_len = strlen (parser->context.sql_user_text);
+      int var_len = 0;
+      bool begin_quote = false;
+
+      host_val = (PARSER_VARCHAR **) malloc (sizeof (PARSER_VARCHAR *) * parser->host_var_count);
+      if (host_val == NULL)
+	{
+	  return ER_OUT_OF_VIRTUAL_MEMORY;
+	}
+
+      for (i = 0; i < parser->host_var_count; i++)
+	{
+	  host_val[i] = pt_print_db_value (parser, &parser->host_variables[i]);
+	  var_len += host_val[i]->length;
+	}
+
+      sbr_text = (char *) malloc (sql_len + var_len);
+      if (sbr_text == NULL)
+	{
+	  return ER_OUT_OF_VIRTUAL_MEMORY;
```

Comment:
if sbr_text is NULL, do you need free(host_val) ?

Reply:
Ok, I will fix it.